### PR TITLE
http base auth added

### PR DESCRIPTION
**Pull request by 0xdeafbeef**

### CODE REVIEW

**Comment by the-c0d3r** [comment]
Hi @rusmonkey thanks for the PR. 
May I know what's the use case for the http basic authentication? This tool is supposed to work on targets where you normally do not have access to any credentials. Similar to the dirbuster in Kali linux, it is intended to be a penetration testing tool to help you bruteforce the URLs. 

If you can provide a solid use case, I will consider merging it, after a few modifications.

**Comment by 0xdeafbeef** [comment]
For example, http://tasks.realctf.pro:37243/, ctf task, where you should bruteforce credentials for basic Auth and ater that you need to find admin panel and bruteforce creds for it

File: admin-finder.py [hunk 1]
```diff
@@ -19,9 +19,10 @@ def main():
 
     logger = setupLogger()
     parser = argparse.ArgumentParser(prog="admin-finder.py", description="Admin panel finder")
-    parser.add_argument("-u", "--url", help="target url/website")
-    parser.add_argument("-w", "--wordlist", help="wordlist to use, default 'wordlist.txt'")
+    parser.add_argument("-u", "--url", help="Target url/website")
+    parser.add_argument("-w", "--wordlist", help="Wordlist to use, default 'wordlist.txt'")
```
Comment (the-c0d3r) [inline admin-finder.py:23]: 👍 for catching the lowercase typo.

File: lib/connection.py [hunk 2]
```diff
@@ -54,20 +59,31 @@ class URLHandler(HTTP):
...
+        else:
+            return self.connect(url, None)["code"]
 
 class RobotHandler(HTTP):
     """Class for handling/analyzing robots.txt"""
-    def __init__(self, url: str) -> None:
+    def __init__(self, url: str, creds) -> None:
+
+        """
+                connect to the url and return the response
```
Comment (the-c0d3r) [inline lib/connection.py]: Please align it to the indentation of `"""`

File: admin-finder.py [hunk 3]
```diff
@@ -19,9 +19,10 @@ def main():
...
     parser.add_argument("-t", "--threadcount", help="Number of threads to use")
+    parser.add_argument("-b", "--baseauth", nargs='+',  help="Basic http auth conditionals")
```
Comment (the-c0d3r) [inline admin-finder.py]:
```suggestion
    parser.add_argument("-c", "--credentials", help="Basic http authentication credentials user:pass format")
```

File: lib/connection.py [hunk 4]
```diff
@@ -17,15 +18,19 @@ def get_headers(self) -> dict:
             "User-Agent": random.choice(self.agents)
         }
 
-    def connect(self, url: str) -> dict:
+    def connect(self, url: str, creds) -> dict:
```
Comment (the-c0d3r) [inline lib/connection.py]:
```suggestion
    def connect(self, url: str, creds: [str, str]) -> dict:
```

File: lib/connection.py [hunk 5]
```diff
@@ -54,20 +59,31 @@ class URLHandler(HTTP):
     def __init__(self) -> None:
         super().__init__()
 
-    def scan(self, url: str) -> int:
+    def scan(self, url: str, creds) -> int:
```
Comment (the-c0d3r) [inline lib/connection.py]:
```suggestion
    def scan(self, url: str, creds: [str, str]) -> int:
```

File: lib/connection.py [hunk 6]
```diff
@@ -54,20 +59,31 @@ class URLHandler(HTTP):
...
-
+        if creds is not None:
+            return self.connect(url, creds)["code"]
+        else:
+            return self.connect(url, None)["code"]
 
 class RobotHandler(HTTP):
     """Class for handling/analyzing robots.txt"""
-    def __init__(self, url: str) -> None:
+    def __init__(self, url: str, creds) -> None:
```
Comment (the-c0d3r) [inline lib/connection.py]:
```suggestion
    def __init__(self, url: str, creds: [str, str]) -> None:
```

File: admin-finder.py [hunk 7]
```diff
@@ -39,9 +40,8 @@ def main():
 
     if args.wordlist is None:
         args.wordlist = "wordlist.txt"
-
     args.url = URLFormatter(args.url).geturl()
```
Comment (the-c0d3r) [inline admin-finder.py:43]:
```suggestion
    args.url = URLFormatter(args.url).geturl()
    
    if args.credentials:
        if ":" not in args.credentials:
            print("[!] Error: credential need to be in this format user:pass")
            exit()
        args.credentials = args.credentials.split(":")
```

File: admin-finder.py [hunk 8]
```diff
@@ -39,9 +40,8 @@ def main():
...
-    robot_handler = RobotHandler(args.url)
+    robot_handler = RobotHandler(args.url, args.baseauth)
```
Comment (the-c0d3r) [inline admin-finder.py]:
```suggestion
    robot_handler = RobotHandler(args.url, args.credentials)
```

File: admin-finder.py [hunk 9]
```diff
@@ -58,7 +58,7 @@ def main():
         workQueue = queue.Queue()
         workerPool = []
         for _ in range(int(args.threadcount)):
-            thread = WorkerThread(workQueue)
+            thread = WorkerThread(workQueue, args.baseauth)
```
Comment (the-c0d3r) [inline admin-finder.py]:
```suggestion
            thread = WorkerThread(workQueue, args.credentials)
```

File: lib/worker.py [hunk 10]
```diff
@@ -7,18 +7,19 @@
 
 class WorkerThread(threading.Thread):
     """The threads that will be processing all the wordlist items"""
-    def __init__(self, queue: queue.Queue) -> None:
+    def __init__(self, queue: queue.Queue, creds) -> None:
```
Comment (the-c0d3r) [inline lib/worker.py]:
```suggestion
    def __init__(self, queue: queue.Queue, creds: [str, str]) -> None:
```

**Review by the-c0d3r — CHANGES_REQUESTED** [review]
That use case you mentioned seem to be only suitable for the specific CTF scenairos. But I intend to modify this tool to be a bit more general purpose URL bruteforcing tool in the future. Thanks for your PR. 

Please make some necessary changes as follows. 
And please remove the unnecessary files as well (eg. `.idea/*`, `__pycache__/*`, `*.pyc` files)

**Comment by 0xdeafbeef** [comment]
@the-c0d3r, all  requested changes are made

File: lib/connection.py (hunk 2 above)
Comment (the-c0d3r) [reply lib/connection.py]: Could you deindent the entire documentation block one or two levels to match the doc string starting indent?

```
    """
    connect to ...
    ...
    """
```

**Review by the-c0d3r — CHANGES_REQUESTED** [review]
Hey, thanks for the effort. One last final change for docstring deindentation.

File: lib/connection.py (hunk 2 above)
Comment (0xdeafbeef) [reply lib/connection.py]: Sorry, it's my first contribution, seems i've fixed everything

**Review by the-c0d3r — APPROVED** [review]

**Comment by the-c0d3r** [comment]
Thanks for the PR @B4DF4C3 and for all those adjustments. I'll merge it now. Feel free to open up more PRs 😄

**Pull request merged**